pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -148,9 +174,22 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 	}
 
 	nextIndex := (b.firstIndex + b.size) % len(b.points)
-	b.points[nextIndex] = p
 	b.size++
-	return nil
+	return &b.points[nextIndex], nil
+}
+
+func (b *HPointRingBuffer) RemoveLastPoint() {
+	if b.size == 0 {
+		return
+	}
+
+	// Remove the last point by adjusting the slice length.
+	b.points = b.points[:len(b.points)-1]
```

Comment:
Why do we need to do this? Can't we just adjust `b.size`? 

This also isn't correct if we've wrapped around to the start of the buffer.